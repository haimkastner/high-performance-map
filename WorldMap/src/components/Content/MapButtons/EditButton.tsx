import React from "react";
import { MapButtonIcon } from "./MapButtonIcon";
import { Box, Button } from "@material-ui/core";
import { setSharedState, subsribeState, unsubsribeState } from "../../../services/stats-store";

interface State {
  speed: string;
}

export class EditButton extends React.Component<any, State> {
  token: any;

  state: State = {
    speed: "111"
  };

  publish() {
    setSharedState("speed", 99);
    setSharedState("direction", 100);
  }

  componentDidMount() {
    this.token = subsribeState("speed", this);
  }
  componentWillUnmount() {
    unsubsribeState(this.token);
  }

  render() {
    return (
      <Box>
        <Button onClick={this.publish}>
          {this.state.speed}
          <MapButtonIcon
            imageName={"display-copy-shape-2@2x.png"}
          ></MapButtonIcon>
        </Button>
      </Box>
    );
  }
}
