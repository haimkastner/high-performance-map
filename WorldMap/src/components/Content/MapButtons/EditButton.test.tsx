import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { EditButton } from './EditButton';

describe('EditButton Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EditButton showMapButtonsMenu={true} speed={"123"} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<EditButton showMapButtonsMenu={true} speed={"123"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('EditButton Component Logic', () => {});
});
