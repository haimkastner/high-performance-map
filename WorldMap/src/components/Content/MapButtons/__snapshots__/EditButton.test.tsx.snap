// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditButton Component matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-37"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <div
        className="MuiBox-root MuiBox-root-70"
      >
        <img
          height={25}
          src="display-copy-shape-2@2x.png"
          width={25}
        />
      </div>
    </span>
  </button>
</div>
`;
